wordpress-mobile/WordPress-Android
# RI: Implement "Turn on/off notifs" action on Reader discover tab

Parent Issue: #12028
 
Adds "Turn on/ off site notifications" action to the items on the discover tab in Reader.

Logic is extracted from `ReaderPostListFragment` to `ReaderSiteNotificationsUseCase` with below fixes:
a) notification status is updated in the `tbl_blog_info` table only on successful subscription update.
b) network error snackbar is displayed if no network is available.

~~**Open Question**~~
~~- Discover feed is not yet updated on updates to `tbl_blog_info`. Should we post `ReaderBlogInfoTableActionEnded.INSTANCE` similar to `ReaderPostTableActionEnded` event?~~ - More menu can be refreshed with the changes in PR #12781 so no need to update the feed.

**To test**

I'd suggest just review the code and run unit tests for now. Turn on/off notifs depend on whether site is followed (it adds an entry to `tbl_blog_info` which is required to update notification status) and FollowUseCase is not merged yet.

We can test it once again when FollowUseCase is finalised.

Notes:

1. Haven't yet used this usecase in the legacy adapter as we update notification status after subscription is updated in this new usecase as compared to optimistic update in the legacy adapter. Plus legacy adapter is buggy - its doesn't care to revert status if request fails. Let me know if I should switch to new usecase there.

2. In case we want to do both optimistic update + update on request completion, we might want to reconsider this design.

PR submission checklist:
